src/icalendar/tests/test_parsing.py

```diff
@@ -0,0 +1,163 @@
+    ('TEL;TYPE=HOME,VOICE:000000000', ('TEL', Parameters({'TYPE': ['HOME', 'VOICE']}), '000000000')),
+    # Issue #143 - Allow dots in property names. Another vCard related issue.
+    # see https://github.com/collective/icalendar/pull/143
+    ('ITEMADRNULLTHISISTHEADRESS08158SOMECITY12345.ADR:;;This is the Adress 08; Some City;;12345;Germany', ('ITEMADRNULLTHISISTHEADRESS08158SOMECITY12345.ADR', Parameters(), ';;This is the Adress 08; Some City;;12345;Germany'))
+])
+def test_content_lines_parsed_properly(raw_content_line, expected_output):
+    assert Contentline.from_ical(raw_content_line).parts() == expected_output
+
+def test_issue_157_removes_trailing_semicolon(events):
+    '''Issue #157 - Recurring rules and trailing semicolons
+
+    https://github.com/collective/icalendar/pull/157
+    '''
+    recur = events.issue_157_removes_trailing_semicolon.decoded("RRULE")
+    assert isinstance(recur, vRecur)
+    assert recur.to_ical() == b'FREQ=YEARLY;BYDAY=1SU;BYMONTH=11'
+
+@pytest.mark.parametrize('calendar_name', [
+    # Issue #178 - A component with an unknown/invalid name is represented
+    # as one of the known components, the information about the original
+    # component name is lost.
+    # https://github.com/collective/icalendar/issues/178 https://github.com/collective/icalendar/pull/180
+    # Parsing of a nonstandard component
+    'issue_178_component_with_invalid_name_represented',
+    # Nonstandard component inside other components, also has properties
+    'issue_178_custom_component_inside_other',
+    # Nonstandard component is able to contain other components
+    'issue_178_custom_component_contains_other'])
+def test_calendar_to_ical_is_inverse_of_from_ical(calendars, calendar_name):
+    calendar = getattr(calendars, calendar_name)
+    assert calendar.to_ical() == calendar.raw_ics
+
+@pytest.mark.parametrize('event_name', [
+    ('issue_100_transformed_doctests_into_unittests'), # https://github.com/collective/icalendar/pull/100
+    ('issue_184_broken_representation_of_period'),
+])
+def test_event_to_ical_is_inverse_of_from_ical(events, event_name):
+    event = getattr(events, event_name)
+    assert event.to_ical() == event.raw_ics
+
+@pytest.mark.parametrize('timezone_file', [
+    # Issue #55 - Parse error on utc-offset with seconds value
+    # see https://github.com/collective/icalendar/issues/55
+    'issue_55_parse_error_on_utc_offset_with_seconds',
+])
+def test_timezones_to_ical_is_inverse_of_from_ical(timezones, timezone_file):
+    timezone = getattr(timezones, timezone_file)
+    assert timezone.to_ical() == timezone.raw_ics
+
+def test_decode_rrule_attribute_error_issue_70(events):
+    # Issue #70 - e.decode("RRULE") causes Attribute Error
+    # see https://github.com/collective/icalendar/issues/70
+    recur = events.issue_70_rrule_causes_attribute_error.decoded('RRULE')
+    assert isinstance(recur, vRecur)
+    assert recur.to_ical() == b'FREQ=WEEKLY;UNTIL=20070619T225959;INTERVAL=1'
+
```

Comment:
I really like the test structure, the different fixtures and the way that there are a few assertions always at the end.